File: subarray-sums-divisible-by-k..java
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        prefix=[0]
        rem={}
        count=0
        for i in nums:
            prefix.append(prefix[-1]+i)
        for i in prefix:
            if i%k in rem:
                count+=rem[i%k]
                rem[i%k]+=1
            else:
                rem[i%k]=1
        return count

File: subarray-sums-divisible-by-k.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        hash_map = defaultdict(int)
        hash_map[0] = 1
        sum_ = 0
        count = 0

        for num in nums:
            sum_ += num
            if sum_ % k in hash_map:
                count += hash_map[sum_ % k]
            hash_map[sum_ % k] += 1
            # hash_map[num % k] += 1

        return count